openhands/integrations/utils.py

```diff
@@ -9,31 +9,47 @@ async def validate_provider_token(
     token: SecretStr, base_domain: str | None = None
 ) -> ProviderType | None:
     """
-    Determine whether a token is for GitHub or GitLab by attempting to get user info
-    from both services.
+    Determine whether a token is for GitHub or GitLab.
+    Checks GitHub /user, then GitHub /installation/repositories, then GitLab /user.
 
     Args:
         token: The token to check
+        base_domain: Optional base domain for GitHub Enterprise or GitLab self-hosted
 
     Returns:
         'github' if it's a GitHub token
         'gitlab' if it's a GitLab token
-        None if the token is invalid for both services
+        None if the token is invalid for all checks
     """
-    # Try GitHub first
+    # 1. Try GitHub /user (PAT or user-scoped token)
     try:
         github_service = GitHubService(token=token, base_domain=base_domain)
         await github_service.get_user()
+        print('GitHub token validated via /user endpoint.')
         return ProviderType.GITHUB
-    except Exception:
-        pass
+    except Exception as e:
+        print(f'GitHub /user check failed: {e}')
 
-    # Try GitLab next
+    # 2. Try GitHub /installation/repositories (GitHub App installation token)
+    try:
+        # Re-initialize or ensure service is still valid if needed
+        # Assuming GitHubService instance is okay after previous failure
+        await github_service.get_app()
+        print('GitHub token validated via /installation/repositories endpoint.')
```

Comment:
Lets use the openhands logger for consistency 

`from openhands.core.logger import openhands_logger as logger`

then use it with 

`logger.warning()` or `logger.error()`